### PR TITLE
test: lib/rstrlen: Added unit tests for lib/rstrlen.

### PR DESCRIPTION
**Pull request by mwierzbix**
Signed-off-by: Michal Jerzy Wierzbicki <michalx.wierzbicki@linux.intel.com>

### CODE REVIEW

File: test/cmocka/Makefile.am [hunk 1]
```diff
@@ -49,6 +49,12 @@ check_PROGRAMS += alloc
 alloc_SOURCES = src/lib/alloc/alloc.c src/lib/alloc/mock.c ../../src/lib/alloc.c ../../src/platform/apollolake/memory.c
 endif
 
+# lib/lib tests
+
+check_PROGRAMS += rstrlen
+rstrlen_SOURCES = src/lib/lib/rstrlen.c
+rstrlen_LDADD = ../../src/lib/libcore_a-lib.o $(LDADD)
```
Comment (jajanusz) [inline test/cmocka/Makefile.am]: I think we shouldn't link to object (.o) files, better to link to lib that you test (.a or .la) file

Comment (mwierzbix) [reply test/cmocka/Makefile.am]: Fixed as per request.

**Comment by lgirdwood** [comment]
@mwierzbix can you fix conflicts. Thanks

**Comment by mwierzbix** [comment]
@lgirdwood conflicts fixed as requested. No probs

**Comment by mwierzbix** [comment]
@lgirdwood This request adds lines to the same `Makefile.am` as https://github.com/thesofproject/sof/pull/203 does , and thanks to the automerge not detecting "relocating" lines, I will have to modify this pull request after https://github.com/thesofproject/sof/pull/203 gets merged. And then squash into single commit.

https://github.com/thesofproject/sof/pull/200
```
1  aaa
2 +bbb
```
https://github.com/thesofproject/sof/pull/203
```
1  aaa
2 +ccc
```

Doesn't become automatically
```
1  aaa
2  bbb
3  ccc
```

I could modify the `Makefile.am` in this pull request to contain the changes introduced in https://github.com/thesofproject/sof/pull/203 , but I think that would be against the spirit of version control? Should I wait for either of them getting merged into master and then update `Makefile.am` in the other, or should I just set the contents of `Makefile.am` to what it would have looked after all my current pull requests modifying this very `Makefile.am`?

**Comment by mwierzbix** [comment]
You see for instance here https://github.com/thesofproject/sof/pull/202/conflicts what happens.

**Comment by jajanusz** [comment]
@mwierzbix @lgirdwood 
Another point why changing buildsystem to CMake would be good - you can help git/github with these conflicts by having makefile per folder for example, however with autotools you would have to not only add makefile for each set of tests but also update configure.ac and we don't want to mess with so many configure files (especially on in the core of the project) for just unit test. With cmake you would just recursivly manage unit tests within UTs domain.

**Comment by lgirdwood** [comment]
@jajanusz we are looking into using kernel build as an update. Linux, uboot, crosstool all use it, we can update build in 1.3 era

**Comment by mwierzbix** [comment]
@lgirdwood @jajanusz Does this pull request requires any changes still?

File: test/cmocka/Makefile.am [hunk 2]
```diff
@@ -55,6 +55,14 @@ check_PROGRAMS += rstrcmp
 rstrcmp_SOURCES = src/lib/lib/rstrcmp.c
 rstrcmp_LDADD = ../../src/lib/libcore.a $(LDADD)
 
+check_PROGRAMS += rstrlen
+rstrlen_SOURCES = src/lib/lib/rstrlen.c
+rstrlen_LDADD = ../../src/lib/libcore.a $(LDADD)
+
+check_PROGRAMS += memset
```
Comment (jajanusz) [inline test/cmocka/Makefile.am]: Are you sure it should be here? Sources for that unit are not in branch.

**Comment by mwierzbix** [comment]
@lgirdwood fixed checkpatch.

**Pull request merged**